Controller/RegistrationFOSUser1Controller.php

```diff
@@ -65,7 +65,7 @@ public function registerAction()
                     $url = $this->get('session')->get('sonata_user_redirect_url');
```

Comment:
How about 
```
$url = $this->get('session')->get(
    'sonata_user_redirect_url', 
    $this->generateUrl('sonata_user_profile_show')
)
``` instead then ? I don't think `$url` can end up being null inside the if, only inside the else…